File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
import java.util.*;
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> set = new HashSet<>();
        HashSet<Integer> result = new HashSet<>();
        for(int n:nums1){
            set.add(n);
        }
        for(int n:nums2){
            if(set.contains(n)){
                result.add(n);
            }
        }
        int[] temp=new int[result.size()];
        int i=0;
        for(int n:result){
            temp[i]=n;
            i++;
        }
        return temp;
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        seen=set()
        container=set()

        for num in nums1:
            if num not in seen:
                seen.add(num)
        for num in nums2:
            if num in seen:
                container.add(num)

        return list(container)